### PR TITLE
use batch idx and node id as unique key for dedup in temporal sampling

### PR DESCRIPTION
**Pull request by yaoyaowd**
This PR would fix the issue where we sample the same node multiple times and blow up batch size.

### CODE REVIEW

**Review by rusty1s — APPROVED** [review]
Wow, this is really cool!

File: csrc/cpu/neighbor_sample_cpu.cpp [hunk 1]
```diff
@@ -223,7 +245,8 @@ hetero_sample(const vector<node_t> &node_types,
       const auto &begin = slice_dict.at(dst_node_type).first;
       const auto &end = slice_dict.at(dst_node_type).second;
       for (int64_t i = begin; i < end; i++) {
-        const auto &w = dst_samples[i];
+        const auto &w = temporal ? temp_dst_samples[i].first : dst_samples[i];
+        const int64_t source_w = temporal ? temp_dst_samples[i].second : -1;
```
Comment (RexYing) [inline csrc/cpu/neighbor_sample_cpu.cpp:249]: why is it called source_w? Is it the root node?

**Review by RexYing — COMMENTED** [review]
The idea looks good to me! I think the code is correct. Look forward to the results after testing! It should save gpu memory for temporal minibatches.

File: csrc/cpu/neighbor_sample_cpu.cpp (hunk 1 above)
Comment (yaoyaowd) [reply csrc/cpu/neighbor_sample_cpu.cpp:249]: yes, do you prefer rename it to `root_w`?

**Review by RexYing — APPROVED** [review]

File: csrc/cpu/neighbor_sample_cpu.cpp (hunk 1 above)
Comment (RexYing) [reply csrc/cpu/neighbor_sample_cpu.cpp]: sounds good

**Pull request merged**